torch/distributed/fsdp/_exec_order_utils.py

```diff
@@ -37,12 +37,8 @@ def __init__(
         # Tracks the (static) pre-forward order for execution order validation
         # and forward prefetching
         self.handles_pre_forward_order: List[FlatParamHandle] = []
-        # Maps each handles key to its index in `handles_pre_forward_order`
-        self.handles_to_pre_forward_order_index: Dict[FlatParamHandle, int] = {}
         # Tracks the post-forward order for pre-backward prefetching
-        self.handles_post_forward_order: List[FlatParamHandle] = []
-        # Maps each handles key to its index in `handles_post_forward_order`
-        self.handles_to_post_forward_order_index: Dict[FlatParamHandle, int] = {}
+        self.handles_post_forward_order: List[Optional[FlatParamHandle]] = []
```

Comment:
IIUC, this is actually a `List[FlatParamHandle]` now because we never append `None` into the list?